src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs

```diff
@@ -120,6 +122,33 @@ End Sub
             }
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        public async Task TestCancellationOnSessionWithSolution()
+        {
+            var code = @"class Test { void Method() { } }";
+
+            using (var workspace = CreateWorkspace(LanguageNames.CSharp, code))
+            {
+                var solution = workspace.CurrentSolution;
+                var solutionChecksum = await solution.State.GetChecksumAsync(CancellationToken.None);
+
+                var source = new CancellationTokenSource();
+                var connection = new MyConnection(source);
+                try
+                {
+                    await SessionWithSolution.CreateAsync(connection, solution, source.Token);
+                }
+                catch (OperationCanceledException)
+                {
+                }
+                
+                // make sure things that should have been cleaned up are cleaned up
+                var service = (RemotableDataServiceFactory.Service)solution.Workspace.Services.GetService<IRemotableDataService>();
+                Assert.Null(service.GetRemotableData_TestOnly(solutionChecksum, CancellationToken.None));
+                Assert.True(connection.Disposed);
```

Comment:
:question: why would this be true here? The ownership transfer isn't clear and the test method doesn't dispose of the object, but I'm also reviewing this on my phone so maybe I just need a better view.

Reply:
whole point of this test is if it throws connection get disposed correctly.